packages/liveblocks-core/src/room.ts

```diff
@@ -1603,9 +1573,33 @@ function makeStateMachine<
     }
   }
 
+  function applyAndSendOps(
+    offlineOps: Map<string, Op>,
+    batchedUpdatesWrapper: (cb: () => void) => void
+  ) {
+    if (offlineOps.size === 0) {
+      return;
+    }
+
+    const messages: ClientMsg<TPresence, TRoomEvent>[] = [];
+
+    const ops = Array.from(offlineOps.values());
+
+    const result = applyOps(ops, true);
+
+    messages.push({
+      type: ClientMsgCode.UPDATE_STORAGE,
+      ops: result.ops,
+    });
+
+    notify(result.updates, batchedUpdatesWrapper);
+
+    effects.send(messages);
+  }
+
   function onMessage(event: MessageEvent<string>) {
     if (event.data === "pong") {
-      clearTimeout(context.timeoutHandles.pongTimeout);
+      clearTimeout(context.timers.pongTimeout);
```

Comment:
If I am understanding this correctly, we don't reschedule here because any activity will mean we can wait for the heartbeat to run again? Shouldn't we also restart the heartbeat interval on pong? Otherwise the heartbeat might happen like 1 second after this and then we've got a ping/pong happening when we don't really need it.

Reply:
I think that's correct, and it would be nice to do that indeed 👍